### PR TITLE
feat(workspace): add uv workspace discovery (TC-4266)

### PR DESCRIPTION
**Pull request by Strum355**
## Summary
- Parse `[tool.uv.workspace]` member and exclude globs from `pyproject.toml` to discover workspace member manifests
- Virtual workspaces (no `[project]` section) correctly exclude root `pyproject.toml` from results
- Add `__pycache__` and `.venv` to default ignore patterns
- Reuse existing `PyprojectTomlUtils` and `WorkspaceUtils` for TOML parsing and ignore filtering

## Test plan
- [x] 7 unit tests covering root-package workspace, virtual workspace, exclude patterns, nested multi-glob patterns, no-lock, no-config, and ignore pattern filtering
- [x] All existing workspace tests continue to pass

[TC-4266](https://redhat.atlassian.net/browse/TC-4266)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[TC-4266]: https://redhat.atlassian.net/browse/TC-4266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by Sourcery

Add support for discovering Python uv workspaces during workspace manifest resolution and extend default ignore patterns.

New Features:
- Discover uv workspace member pyproject.toml files based on [tool.uv.workspace] configuration and uv.lock presence.
- Support both root-package and virtual uv workspaces when resolving workspace manifests.

Enhancements:
- Extend default workspace discovery ignore patterns to skip __pycache__ and .venv directories.

Tests:
- Add unit test coverage for uv workspace discovery including root, virtual, excluded, nested, missing-lock, missing-config, and ignore-pattern scenarios.

### CODE REVIEW

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java [hunk 1]
```diff
@@ -985,6 +1007,87 @@ private List<Path> discoverGoWorkspaceModules(Path workspaceDir, Set<String> ign
...
+                  boolean matchesExclude =
+                      excludeMatchers.stream().anyMatch(m -> m.matches(relative));
+                  if (matchesMember && !matchesExclude) {
+                    manifests.add(p);
+                  }
+                });
+      }
+
+      if (PyprojectTomlUtils.getProjectName(toml) != null) {
+        manifests.addFirst(rootPyproject);
```
[the remark this answers is not in the document]
Comment (Strum355) [reply src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java:1063]: **Classification: False Positive** -- `ArrayList.addFirst()` is valid in Java 21+ (this project uses `maven.compiler.release=21`). Java 21 introduced the `SequencedCollection` interface, which `ArrayList` implements, providing `addFirst()`. The same pattern is used elsewhere in this file (line 910). No change needed.

*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11.*

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java [hunk 2]
```diff
@@ -985,6 +1007,87 @@ private List<Path> discoverGoWorkspaceModules(Path workspaceDir, Set<String> ign
...
+      }
+
+      return WorkspaceUtils.filterByIgnorePatterns(workspaceDir, manifests, ignorePatterns);
+    } catch (Exception e) {
+      LOG.warning("Failed to discover uv workspace members: " + e.getMessage());
```
[the remark this answers is not in the document]
Comment (Strum355) [reply src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java]: **Classification: Resolved** -- This suggestion was implemented in commit b3a48df. The catch block now uses `LOG.log(Level.WARNING, "Failed to discover uv workspace members", e)` preserving the full stack trace. The `java.util.logging.Level` import is present at line 72.

*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11.*

File: src/test/java/io/github/guacsec/trustifyda/impl/UvWorkspaceDiscoveryTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  @Test
+  void discoverWorkspaceManifests_uvRootPackageWorkspace() throws IOException {
+    Path workspaceDir = UV_FIXTURES.resolve("uv_workspace").toAbsolutePath().normalize();
+
+    ExhortApi api = new ExhortApi(Mockito.mock(java.net.http.HttpClient.class));
+    List<Path> manifests = api.discoverWorkspaceManifests(workspaceDir, Set.of());
+
+    assertThat(manifests).hasSize(3);
+    assertThat(manifests).allMatch(p -> p.toString().endsWith("pyproject.toml"));
+    assertThat(manifests.getFirst()).isEqualTo(workspaceDir.resolve("pyproject.toml"));
```
[the remark this answers is not in the document]
Comment (Strum355) [reply src/test/java/io/github/guacsec/trustifyda/impl/UvWorkspaceDiscoveryTest.java:42]: **Classification: Nice-to-have** -- The suggestion to add a test for whitespace-only members is a reasonable edge case test. The current code already handles this correctly (the `isBlank()` check filters them out and `memberPatterns.isEmpty()` returns early). This is a test coverage improvement, not a bug fix. Could be addressed in a follow-up.

*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11.*

**Comment by Strum355** [comment]
## Verification Report — TC-4266

**PR:** #446 | **Branch:** `TC-4266` → `main` | **Verdict:** PASS

### Review Feedback
| # | Thread | Classification | Action |
|---|--------|---------------|--------|
| 1 | sourcery-ai: ArrayList.addFirst() bug risk | False positive (Java 21+ API) | No action |
| 2 | sourcery-ai: logging suggestion | Resolved (commit b3a48df) | No action |
| 3 | sourcery-ai: empty members test suggestion | Nice-to-have | No action |

All 3 threads classified in prior run; no new threads.

### CI Status
**46/46 checks PASS**

### Scope Containment
Single clean commit `8602496` — uv workspace discovery only.
Files modified: `ExhortApi.java` (uv detection + `discoverUvWorkspaceMembers()` method), `WorkspaceUtils.java` (glob fix for root-level paths), `UvWorkspaceDiscoveryTest.java`, plus 15 test fixture files.

**Previously:** PR had wrong base branch (TC-4264) and included Go workspace commits. Fixed by rebasing onto `main` and squashing to a single uv-only commit. Base branch updated to `main`.

### Sensitive Patterns
No secrets, credentials, or API keys detected.

### Acceptance Criteria
| # | Criterion | Status |
|---|-----------|--------|
| 1 | `stackAnalysisBatch()` discovers all member `pyproject.toml` files | PASS |
| 2 | TOML parsing extracts `[tool.uv.workspace].members` and `.exclude` | PASS |
| 3 | Glob expansion of member patterns works with Java PathMatcher | PASS |
| 4 | Virtual workspaces exclude root from manifests | PASS |
| 5 | Root-package workspaces include root in manifests | PASS |
| 6 | `**/__pycache__/**` and `**/.venv/**` added to default ignore patterns | PASS |
| 7 | Missing `uv.lock` → not detected as workspace | PASS |

### Test Quality
7 tests in `UvWorkspaceDiscoveryTest.java` — all pass. Covers: simple workspace, virtual workspace (root excluded), root-package workspace (root included), exclude patterns, nested member discovery, no workspace config, and missing uv.lock.

### Verification Commands
| Command | Result |
|---------|--------|
| `mvn test -Dtest=UvWorkspaceDiscoveryTest` | PASS (7 tests, 0 failures) |

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11. Updated after branch cleanup (rebase onto main, single clean commit).*

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java [hunk 4]
```diff
@@ -930,6 +946,87 @@ private List<Path> discoverCargoManifests(Path workspaceDir, Set<String> ignoreP
...
+        return Collections.emptyList();
+      }
+
+      TomlArray membersArray = workspaceConfig.getArray("members");
+      if (membersArray == null || membersArray.isEmpty()) {
+        return Collections.emptyList();
+      }
+
+      List<String> memberPatterns = new ArrayList<>();
+      for (int i = 0; i < membersArray.size(); i++) {
```
Comment (a-oren) [inline src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java]: The TomlArray-to-string extraction loop appears twice (for members and exclude) with identical null/blank filtering. Consider extracting a small helper

```suggestion
      static List<String> toStringList(TomlArray array) {                                                                                             
      if (array == null) return List.of();                                                                                                        
      List<String> result = new ArrayList<>();                                                                                                    
      for (int i = 0; i < array.size(); i++) {                                                                                                    
          String s = array.getString(i);                                                                                                          
          if (s != null && !s.isBlank()) {                                                                                                        
              result.add(s.trim());                                                                                                               
          }                                                                                                                                       
      }                                                                                                                                           
      return result; 
```

**Review by a-oren — APPROVED** [review]
LGTM
